### PR TITLE
Make connection and color packages peer dependencies

### PR DESCRIPTION
**Pull request by xtinec**
🐛 Bug Fix
- Also updating the version of the color package in the demo package to be the latest

### CODE REVIEW

File: packages/superset-ui-chart/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "react": "^15 || ^16"
   },
   "peerDependencies": {
+    "@superset-ui/connection": "^0.7.0",
```
Comment (xtinec) [inline packages/superset-ui-chart/package.json]: This is actually a precursor to the embeddable demo PR but there is no harm adding this now. 🙂

Comment (xtinec) [reply packages/superset-ui-chart/package.json]: why?

File: packages/superset-ui-demo/package.json [hunk 2]
```diff
@@ -33,11 +33,13 @@
...
     "@storybook/react": "^4.0.2",
-    "@superset-ui/color": "^0.4.0",
     "babel-loader": "^8.0.4",
     "git-directory-deploy": "^1.5.1",
     "react": "^16.6.0",
     "react-dom": "^16.6.0",
     "storybook-addon-jsx": "^5.4.0"
+  },
+  "peerDependencies": {
+    "@superset-ui/color": "^0.7.0"
```
[the remark this answers is not in the document]
Comment (xtinec) [reply packages/superset-ui-demo/package.json]: Wait, shouldn't the `color` package be a peer dependency of (the word-cloud package)? I should probably leave this one as-is (color is a runtime dependency of the demo package) till we add the word-cloud package as a runtime dependency.

File: packages/superset-ui-chart/package.json (hunk 1 above)
Comment (kristw) [reply packages/superset-ui-chart/package.json]: For local development, `yarn install` or `npm install` don't install `peerDependencies` so adding them to `devDependencies` so it is installed automatically.

File: packages/superset-ui-demo/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (kristw) [reply packages/superset-ui-demo/package.json]: But this file is the `demo`'s `package.json`?

[the remark this answers is not in the document]
Comment (xtinec) [reply packages/superset-ui-demo/package.json]: Yeah, I think I jumped the gun here adding color to peerDependency. It should only be needed when we add word-cloud package here. I'll revert this change here.

**Pull request merged**